apps/prairielearn/assets/scripts/instructorAssessmentInstancesClient.tsx

```diff
@@ -401,93 +405,325 @@ onDocumentReady(() => {
+            id="reopen_with_limit"
+            value="false"
+            {...(reopenWithoutLimit !== 'true' ? { checked: true } : {})}
+            onClick={(e) => handleChange(e)}
+          />
+          <label class="form-check-label" for="reopen_with_limit">
+            Re-open with time limit
+          </label>
+        </div>
+      </div>
+    );
+  }
+
+  function TimeLimitSelect({
+    time_remaining_sec,
+    has_open_instance,
+    open,
+    handleChange,
+  }: {
+    time_remaining_sec: number | null;
+    has_open_instance: boolean;
+    open: boolean;
+    handleChange: (e: Event) => void;
+  }) {
+    const showRemoveTimeLimit = !open || (time_remaining_sec !== null && time_remaining_sec > 0);
+    return (
+      <select
+        class="custom-select select-time-limit"
+        name="plus_minus"
+        aria-label="Time limit options"
+        onChange={(e) => handleChange(e)}
+      >
+        {time_remaining_sec !== null
+          ? has_open_instance
+            ? [
+                <option value="+1">Add to instances with time limit</option>,
+                <option value="-1">Subtract from instances with time limit</option>,
+              ]
+            : [<option value="+1">Add</option>, <option value="-1">Subtract</option>]
+          : ''}
+        <option value="set_total">Set total time limit</option>
+        <option value="set_rem">Set remaining time</option>
+        <option value="set_exact">Set exact closing time</option>
+        {showRemoveTimeLimit ? <option value="unlimited">Remove time limit</option> : ''}
+        {showRemoveTimeLimit ? <option value="expire">Expire time limit</option> : ''}
+      </select>
+    );
+  }
+
+  function TimeAddInput({
+    timeAdd,
+    handleChange,
+  }: {
+    timeAdd: number;
+    handleChange: (e: Event) => void;
+  }) {
+    return (
+      <div class="input-group">
+        <input
+          class="form-control time-limit-field"
+          type="number"
+          name="time_add"
+          aria-label="Time value"
+          value={timeAdd}
+          onChange={(e) => handleChange(e)}
+        />
+        <span class="input-group-text time-limit-field">minutes</span>
+      </div>
+    );
+  }
+
+  function DatePicker({ handleChange, date }: { handleChange: (e: Event) => void; date: string }) {
+    return (
+      <div class="input-group date-picker">
+        <input
+          class="form-control date-picker"
+          type="datetime-local"
+          name="date"
+          value={date}
+          onChange={(e) => handleChange(e)}
+        />
+        <span class="input-group-text date-picker">{timezone}</span>
+      </div>
+    );
+  }
+
+  function TimeLimitExplanation({ plusMinus }: { plusMinus: string }) {
+    let explanation = '';
+    if (plusMinus === 'set_total') {
+      explanation =
+        'Updating the total time limit will set the given amount of time for the assessment based on when the assessment was started.';
```

Comment:
The varying message lengths here cause the width of the popover to jump all over the place. You probably want something like `width: XXXpx, max-width: 90vw` to prevent that.

Reply:
That makes sense. I tried adding `width: 400px, max-width: 90vw` here but the issue was that the display was inline. Added a `.d-block` class to the explanation and now the explanation text wraps rather than expanding the width of the popover.

Reply:
Added a new class `popover-narrow-fixed` with `width: 400px` and `max-width: 90vw` in [42543c9](https://github.com/PrairieLearn/PrairieLearn/pull/10800/commits/42543c9a1ca769d8b9f72a66c1f6d1916060f477).